### PR TITLE
Vendor gomobile

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers @gabriel (@maxtaco to make sure this vendoring looks okay) 

This vendors gomobile so that we build our gomobile tool from a specific version so we don't have it break or change underneath us.

This breaking has happened several times before and most recently from this commit: https://github.com/golang/mobile/commit/2f75be449fbada35ca42f481f9878ba7f02a7e7d

for a meaningfull diff just look at the first commit: https://github.com/keybase/client/commit/5ceadb1a6153d63aaa2848fc5368b77d5dbed02e

### CODE REVIEW

File: react-native/gobuild.sh [hunk 1]
```diff
@@ -59,10 +60,16 @@ rsync -pr --ignore-times "$go_client_dir/vendor" "$GOPATH/src/github.com/keybase
...
+vendor_path="$GOPATH/src/github.com/keybase/vendor"
+gomobile_path="$vendor_path/golang.org/x/mobile/cmd/gomobile"
 
-if [ ! -f "$GOPATH/bin/gomobile" ]; then
-  echo "Build gomobile..."
-  go get golang.org/x/mobile/cmd/gomobile
+echo "Build gomobile..."
+(cd "$gomobile_path" && go build -o "$GOPATH/bin/gomobile")
+# The gomobile binary only looks for packages in the GOPATH,
+rsync -pr --ignore-times "$vendor_path/" "$GOPATH/src/"
```
Comment (chromakode) [inline react-native/gobuild.sh]: Is it possible to add another value to the `$GOPATH` env var instead of copying the files?

Comment (chromakode) [inline react-native/gobuild.sh]: This is a :sheep:

**Comment by chromakode** [comment]
:+1:

**Comment by gabriel** [comment]
I wonder if we want to vendor a build tool.. 

We could git clone the mobile repo (at a revision) into the TMP_GO_PATH instead of vendoring?

**Comment by gabriel** [comment]
I'm on with vendoring, the repo looks pretty small

**Comment by gabriel** [comment]
👍

**Pull request merged**